### PR TITLE
Register user

### PR DESCRIPTION
**Pull request by cmsigrist**
User's informations can now be set in the firestore database

### CODE REVIEW

**Comment by cgntnr** [comment]
Hey @cmsigrist , thanks for adding Scroll View to the Passions Page it seems that the tests pass on Cirrus as well. Great job! Just wanted to warn you about the position of button below which is different than other setup pages.

**Comment by cmsigrist** [comment]
> Hey @cmsigrist , thanks for adding Scroll View to the Passions Page it seems that the tests pass on Cirrus as well. Great job! Just wanted to warn you about the position of button below which is different than other setup pages.

Thanks! Yep, it's noted I'll correct this in the next commit :)

File: app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt:42]: I don't see why plural form is converted to singular, here we say e.g. "Show me men" in general and not a specific man or a woman so it was more fitting. Same remark for test class as well.

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGenderMore.kt
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGenderMore.kt:20]: Why this was removed, if regex doesn't match in line 30 warning2_p4_2 is shown. If on the second try an empty text is given as input warning1_p4_2 is shown with line 44. By deleting this line you would make them both stay on page superposed.

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt:27]: A little typo on Orientation.

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt:42]: Reminder for same typo : Orientation.

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/PageViewModel.kt
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/ui/main/PageViewModel.kt:10]: Exactly! Not sure if this is used anywhere.

File: app/src/main/res/values/strings.xml
Comment (cgntnr) [inline app/src/main/res/values/strings.xml:60]: Again I don't see why this is in singular form.

**Review by cgntnr — COMMENTED** [review]
Goal oriented and nicely done. Many refactoring are done along the way as well. Still I added some comments you should check before merging.

File: app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt:42]: I changed it to match the form in ProfileShowMe, but it probably made more sense to do it the other way around (and modify ProfileShowMe...)
We could maybe change all of these in the next refractor (we'll probably need to change the profile setups to fragments)?

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGenderMore.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGenderMore.kt:20]: Yep my mistake! It's back in the next commit :)

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt:27]: Oh no! Thanks :)

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/PageViewModel.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/ui/main/PageViewModel.kt:10]: I may have used it without completely understanding how it worked, when setting up the different tabs for the main screen with fragments...  :/
Maybe I could look into it in the next sprint, and maybe refractor the profile setup along the way?

File: app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt:42]: Yeah grammatically and in terms of what the buttons show ProfileShowMe should have been changed. In the end it's as you wish but this PR changed the other way around and if we leave this to future PR/refactor they can easily become a could-have and we might forgot/miss all the places you have changed.

Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt]: Yep you're right, I think I caught 'em all! ;-)

**Review by cgntnr — APPROVED** [review]
All good, thanks!

File: app/build.gradle
Comment (Xelowak) [inline app/build.gradle:81]: Cirrus says that the version 1.3.0 is available, might be nice to update !

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGenderMore.kt
Comment (Xelowak) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGenderMore.kt:70]: Might be a good idea to import these functions directly so that you don't need to write `ViewAssertions`, `ViewMatchers` and `Matchers` anymore. (We did it in other test classes)

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileOrientation.kt
Comment (Xelowak) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileOrientation.kt:94]: Shouldn't it be a const val outside of the class ? (Like in `TestProfileBirthday`)

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileBirthday.kt
Comment (Xelowak) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileBirthday.kt:24]: I believe this val should be private.

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGender.kt
Comment (Xelowak) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGender.kt:25]: I think these ones should be private too.

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt:41]: It would be better to create a private const val like in other classes. The same applies for the next 2 cases in the code.

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileShowMe.kt
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileShowMe.kt:44]: Cirrus warning about the fact that using the default locale might cause bugs. If I understood it well, I think it is better to write `capitalize(myString)` instead of `myString.capitalize()`.

File: app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt:72]: Is there something that doesn't work ?

File: app/src/main/res/values/strings.xml
Comment (Xelowak) [inline app/src/main/res/values/strings.xml:112]: Some added strings seem to not be used anywhere (Cirrus warnings).

**Review by Xelowak — APPROVED** [review]
It's ok for me. Just look at the comments I made. I think it might be better to be consistant about the constants in classes (are they private ? before the class declaration ? are they const ?) because it isn't the same in every class.

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGenderMore.kt
Comment (cmsigrist) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGenderMore.kt:70]: I didn't write this piece of code (but I guess it considered it to be modified when I merged the main back into my branch).
But yes it would be nice if later when we refractor someone could clean up all the different imports in the tests

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGender.kt
Comment (cmsigrist) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGender.kt:25]: Should they? They're actually shared in multiple tests classes (SettingsShowMeTest and SettingsTest for example). I'm not sure if it's better to copy them all over our tests or simply use the same evreywhere.

File: app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt:72]: This one uses a deprecated function, I just added a //TODO when I came across this

File: app/src/main/res/values/strings.xml
Comment (cmsigrist) [reply app/src/main/res/values/strings.xml:112]: Right I'll remove it in my next branch (I don't have the energy to wait another 25 minutes for cirrus to build)

File: app/build.gradle
Comment (cmsigrist) [reply app/build.gradle:81]: Right I'll change that in the next branch I push :)

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileOrientation.kt
Comment (cmsigrist) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileOrientation.kt:94]: Yep took not of it, it's in the next branch fix

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt:41]: You mean, one private const val: man, woman and more or a var?

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGender.kt
Comment (Xelowak) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGender.kt:25]: My bad, I didn't see that they were used somewhere else. Better let them public !

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt
Comment (Xelowak) [reply app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt:41]: It is probably clearer if they are separated in 3 distinct const vals.

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGenderMore.kt
Comment (cgntnr) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/TestProfileGenderMore.kt:70]: In refactoring #97  I've done those imports for usual classes but not for the test classes. For normal classes it was more important in terms of usability and readability I believe, but it's possible to do in test classes as well. Though certainly not an urgent issue.

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileGender.kt:41]: Okay! Again, I'll push it in the next branch (there's some refactoring that will touch the same class anyway)

**Pull request merged**
